### PR TITLE
Initial orders implimentation and cleanup

### PR DESCRIPTION
**Pull request by morloderex**
- Checkout unit test has been removed
- Removed serval OrderRepository methods as we are now using factory pattern for these.
- Add new initial API and route for creating orders (needs more work for stuff like quantity checking etc).
- Renamed OrderProduct Model to OrderItem.
- Renamed ecom_order_product table to ecom_order_item
- Removed Exceptions from OrdersRepository
- Removed method that was used to make invoices (can be reintroduced but it was wrong)
- Removed OrderDetailsUnitTest class as all of the tests in here was out of date with this new implementation
- Flattened product relation and customer relation on the orders tables and order item tables
- Removed checkout repository as it was out of date
- Added validation rules classes for order_item and orders
- Renamed OrderCreateEvent to OrderCreatedEvent

Signed-off-by: Michael Lundbøl <ml@napp.dk>

### CODE REVIEW

File: src/Orders/Controller/API/OrdersController.php [hunk 1]
```diff
@@ -31,50 +33,43 @@ public function __construct(OrderActions $orderActions, OrderTransformerInterfac
...
-        $data = [
-            'order' => $order,
-            'products' => $order->products,
-            'customer' => $order->customer,
-            'courier' => $order->courier,
-            'address' => $this->transformAddress($order->address),
-            'status' => $order->orderStatus,
-            'payment' => $order->paymentMethod
-        ];
+        return \DB::connection(config('database.default'))->transaction(function () use ($request, $transformer) {
```
Comment (viezel) [inline src/Orders/Controller/API/OrdersController.php]: Please wrap this in a repository method

File: src/Orders/Actions/OrderActions.php [hunk 2]
```diff
@@ -61,16 +76,69 @@ public function index($order = 'created_at', $sort = 'desc')
...
+        event(new OrderCreatedEvent($order, $orderItems));
+
+        return $order->loadMissing('items');
+    }
+
+    /**
+     * @param array $orderItems
+     * @return int
+     */
+    protected function calculateTotalPrice(array $orderItems): int
```
Comment (morloderex) [inline src/Orders/Actions/OrderActions.php]: This is a very naive implimentation of the total order price. It doesn't take into account any discount on the order item or on the order itself.

File: config/laracom.php [hunk 3]
```diff
@@ -78,6 +78,12 @@
         \Laracommerce\Core\Countries\Transformations\CountryTransformerInterface::class => Laracommerce\Core\Countries\Transformations\CountryTransformer::class,
         \Laracommerce\Core\Warehouse\Transformations\WarehouseTransformerInterface::class => Laracommerce\Core\Warehouse\Transformations\WarehouseTransformer::class,
         \Laracommerce\Core\Countries\Transformations\CountryTransformerInterface::class => Laracommerce\Core\Countries\Transformations\CountryTransformer::class,
+    ],
+
+    'shop' => [
+        'orders' => [
+            'check_stock_lookup' => false
```
Comment (viezel) [inline config/laracom.php]: rename to `manage_stock`

Comment (morloderex) [reply config/laracom.php]: Great idea.

File: src/Orders/Actions/OrderActions.php [hunk 4]
```diff
@@ -61,16 +85,60 @@ public function index($order = 'created_at', $sort = 'desc')
...
+
+        $customer = $this->customerRepository->find($request->input('customer')['id']);
+
+        if (null === $customer) {
+            throw new CustomerNotFoundException('Customer not found');
+        }
+
+        $items = $this->productRepository->findMany('sku', array_pluck($requestedItems, 'sku'));
+
+        $orderItems = $this->orderRepository->createOrderDetails($requestedItems, $items, $request->input('store_id'));
```
Comment (viezel) [inline src/Orders/Actions/OrderActions.php]: Change to Warehouse_id

**Pull request merged**